### PR TITLE
Replace GIFs with new GIF viewer

### PR DESCRIPTION
**Pull request by joshuatbrown**
## Issues covered
#1168 

## Description
Like it says in the title. GIFs now appear in the feed, profile, and thread with a "GIF" button on top of them. 

## How to test
1. Navigate to Settings
2. Toggle Enable new media display to "On"
3. Find or post a GIF (here's a note with one: note1nvf6wm257sku94zt9dqaw439a67dg6qq8wwjh9xg2pny07rpvk8qp8xkcq)
4. Tap anywhere on the GIF to play the animation
5. Tap a GIF that's animating to view the GIF full screen
6. Double-tap to zoom in the full-screen viewer
7. Double-tap to zoom out
8. Tap the "X" button or swipe down to close the image viewer

## Screenshots/Video
https://github.com/user-attachments/assets/37e44ea1-af2b-4989-9bb6-1f3a0cddeed9

### CODE REVIEW

File: Nos.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3232,8 +3232,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/SDWebImage/SDWebImageSwiftUI";
 			requirement = {
-				kind = upToNextMajorVersion;
-				minimumVersion = 2.0.0;
+				branch = bugfix/webimage_isAnimating_binding;
```
Comment (joshuatbrown) [inline Nos.xcodeproj/project.pbxproj:3235]: We'll definitely want to update this before merging the PR. But I thought there's a chance they might get the [SDWebImageUI PR](https://github.com/SDWebImage/SDWebImageSwiftUI/pull/333) merged and released before _this_ PR is reviewed and ready to merge. Either way, I'll set this to a released version of SDWebImageSwiftUI before merging -- the only question is whether I'll have to use a workaround in our code or not.

Comment (joshuatbrown) [reply Nos.xcodeproj/project.pbxproj]: Updated -- we're on release 3.1.2 now!

**Review by mplorentz — CHANGES_REQUESTED** [review]
This code is so clean, I love it. I did find some weirdness when rotating gifs. See the two below cases:

- I expect this gif to fill the screen vertically when in landscape, but there is a gap at the bottom: `note1gwug9ke6x40uk8qku0fy0eszr7jldfwj4ap6meqzqhcr3jvyg7kswym5zs` 

https://github.com/user-attachments/assets/715fd4cd-68db-4cac-b676-99aab06034f5

- This gif has a weird gap in portrait after I pinch to zoom a bit in landscape: `note19h0x9um6ymwnmmnjqnc6mvmfpcnqq0t0qmam009fwdlwnzlmrqgq76aptp` 

https://github.com/user-attachments/assets/b80e2b45-fc64-4c0c-bc46-261bf313f6b0

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added a feature flag toggle for “Enable new media display” to Staging builds. [skip-release-notes]
 - Added a new gallery view to display multiple links in a post. Currently behind the “Enable new media display” feature flag. [skip-release-notes]
 - Show single images and gallery view in the proper orientation. Currently behind the “Enable new media display” feature flag. [skip-release-notes]
+- Added an overlay to GIFs that plays the animation when tapped. Currently behind the “Enable new media display” feature flag. [skip-release-notes]
```
[the remark this answers is not in the document]
Comment (joshuatbrown) [reply CHANGELOG.md:23]: That's the idea. Right now it's 100% manual; maybe in the future it'll be 100% automated. And maybe #1438 will change the approach slightly.

**Comment by joshuatbrown** [comment]
> This code is so clean, I love it. I did find some weirdness when rotating gifs. See the two below cases:
> 
> * I expect this gif to fill the screen vertically when in landscape, but there is a gap at the bottom: `note1gwug9ke6x40uk8qku0fy0eszr7jldfwj4ap6meqzqhcr3jvyg7kswym5zs`
> 
> * This gif has a weird gap in portrait after I pinch to zoom a bit in landscape: `note19h0x9um6ymwnmmnjqnc6mvmfpcnqq0t0qmam009fwdlwnzlmrqgq76aptp`
 
Unfortunately I think these are general issues with our ImageViewer; the same weird things happen for me when rotating plain old images: `note1qdsj4qdn6vy5sfmpn93wp4tpx6czggcrthckagz33mdkja8a68pqd0mjsg`

I'll look and see what I can do.

**Comment by mplorentz** [comment]
@joshuatbrown 
> Unfortunately I think these are general issues with our ImageViewer; the same weird things happen for me when rotating plain old images: `note1qdsj4qdn6vy5sfmpn93wp4tpx6czggcrthckagz33mdkja8a68pqd0mjsg`
> 
> I'll look and see what I can do.

In that case I approve and we can address that in another PR.

**Review by mplorentz — APPROVED** [review]

**Comment by joshuatbrown** [comment]
> > Unfortunately I think these are general issues with our ImageViewer; the same weird things happen for me when rotating plain old images: `note1qdsj4qdn6vy5sfmpn93wp4tpx6czggcrthckagz33mdkja8a68pqd0mjsg`
> > I'll look and see what I can do.
> 
> In that case I approve and we can address that in another PR.

Yeah, I ended up debugging a bit and documenting my findings in a Notion doc. I'll create a ticket, add my notes to it, and we can address it when it's prioritized.

**Comment by joshuatbrown** [comment]
New ticket to address issues: https://github.com/planetary-social/nos/issues/1440

**Pull request merged**